hera_cal/io.py

```diff
@@ -864,6 +864,147 @@ def iterate_over_times(self, Nints=1, times=None):
             yield self.read(times=times[i:i + Nints])
 
 
+def read_hera_hdf5(filenames, bls=None, pols=None, full_read_thresh=0.002,
+                   read_data=True, read_flags=False, read_nsamples=False, check=False,
+                   dtype=np.complex128, verbose=False):
+    '''A ~100x faster interface to getting data out of HERA HDF5 files. Only concatenates
+    along time axis. Assumes that if antenna_numbers header array stays the same,
+    the baseline order in the files stays the same. Puts times in ascending order,
+    but does not check that files are contiguous.
+
+    Arguments:
+        filenames: list of files to read
+        bls: list of (ant_1, ant_2, [polstr]) tuples to read out of files
+        pols: list of pol strings to read out of files
+        full_read_thresh (0.01): fractional threshold for reading whole file 
+                                 instead of baseline by baseline.
+        read_data (bool, True): read visdata
+        read_flags (bool, False): read flags
+        read_nsamples (bool, False): read nsamples
+        check (bool, False): run sanity checks to make sure files match.
+        dtype (np.complex128): numpy datatype for output complex-valued arrays
+        verbose: print some progress messages.
+
+    Returns:
+        rv: dictionary with keys 'info' (metadata), 'visdata' (dictionary of waterfalls
```

Comment:
Also, can you list what's in the metadata and what formats to expect in there?

Reply:
And say whether the metadata is relevant to the whole file or to the selection that was performed? For example, if you ask for one baseline, will info know about all the other baselines?

Reply:
Done